### PR TITLE
Widgets: Data formatting

### LINKED ISSUE #7565: [Widgets] Render store info data
# Description

After the store info data is fetched and stored, let's render it in the widget.

# Tasks

- [ ] Render stored info data
- [ ] Make sure data is formatted with the correct currency settings

# UI 

Image

### PR DESCRIPTION
**Pull request by ealeksandrov**
Closes: #7565
Depends on #7707 (not logic, it's just to prevent merge conflicts in same files).

# Description
We need store currency setting to correctly display amount field. This PR also fixes conversion field from overflow over 100%.

# How

- Updates conversion rate value to behave the same way as on dashboard.
- Adds Codable conformance to `CurrencySettings` and shares it between app and extension.
- Uses shared settings to correctly format Amount value.

# Testing Steps

- Set currency settings to be very different from normal. In `wp-admin` it's in WooCommerce -> Settings -> Currency Options.
- Launch the app.
- Compare amount field formatting on app dashboard and widget.

# Screenshots

USD | EUR
--|--
Simulator Screen Shot - 1|Simulator Screen Shot - 2

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/StoreWidgets/StoreInfoProvider.swift [hunk 1]
```diff
@@ -53,16 +54,20 @@ final class StoreInfoProvider: TimelineProvider {
...
+    private let reloadInterval: TimeInterval = 30 * 60
+
     /// Redacted entry with sample data.
     ///
     func placeholder(in context: Context) -> StoreInfoEntry {
         let dependencies = Self.fetchDependencies()
         return StoreInfoEntry.data(.init(range: Localization.today,
                                          name: dependencies?.storeName ?? Localization.myShop,
-                                         revenue: "$132.234",
+                                         revenue: formattedAmountString(for: 132.234, with: dependencies?.storeCurrencySettings),
```
Comment (Ecarrion) [inline WooCommerce/StoreWidgets/StoreInfoProvider.swift]: Neat!

File: WooFoundation/WooFoundation/Currency/CurrencySettings.swift [hunk 2]
```diff
@@ -394,4 +394,32 @@ public class CurrencySettings {
...
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(currencyCode, forKey: .currencyCode)
+        try container.encode(currencyPosition, forKey: .currencyPosition)
+        try container.encode(groupingSeparator, forKey: .groupingSeparator)
+        try container.encode(decimalSeparator, forKey: .decimalSeparator)
+        try container.encode(fractionDigits, forKey: .fractionDigits)
+    }
```
Comment (Ecarrion) [inline WooFoundation/WooFoundation/Currency/CurrencySettings.swift:428]: Since we have a custom encoding here, maybe its worth noting that it was added for the widget extension. So future devs will be more informed.

File: WooCommerce/StoreWidgets/StoreInfoProvider.swift [hunk 3]
```diff
@@ -129,14 +133,41 @@ private extension StoreInfoProvider {
...
+                            storeID: storeID,
+                            storeName: storeName,
+                            storeTimeZone: storeTimeZone,
+                            storeCurrencySettings: storeCurrencySettings)
     }
 }
 
 private extension StoreInfoProvider {
+
+    func formattedAmountString(for amountValue: Decimal, with currencySettings: CurrencySettings?) -> String {
```
Comment (Ecarrion) [inline WooCommerce/StoreWidgets/StoreInfoProvider.swift]: I think these could be static, right? Since they don't access `self` at all.

File: WooCommerce/StoreWidgets/StoreInfoProvider.swift [hunk 4]
```diff
@@ -129,14 +133,41 @@ private extension StoreInfoProvider {
...
+
+    func formattedConversionString(for conversionRate: Double) -> String {
+        let numberFormatter = NumberFormatter()
+        numberFormatter.numberStyle = .percent
+        numberFormatter.minimumFractionDigits = 1
+
+        // do not add 0 fraction digit if the percentage is round
+        let minimumFractionDigits = floor(conversionRate * 100.0) == conversionRate * 100.0 ? 0 : 1
+        numberFormatter.minimumFractionDigits = minimumFractionDigits
+        return numberFormatter.string(from: conversionRate as NSNumber) ?? Constants.valuePlaceholderText
```
Comment (Ecarrion) [inline WooCommerce/StoreWidgets/StoreInfoProvider.swift:160]: I think there is a bit of an edge case here or in the conversion formula as because `100%` is showing as `10.000%`
IMG_718F09EAB50E-1

In retrospective, I should have added unit tests to these types 🙈

**Review by Ecarrion — COMMENTED** [review]
Just a minor issue in the conversion rate and then we should we good to go!

File: WooFoundation/WooFoundation/Currency/CurrencySettings.swift (hunk 2 above)
Comment (ealeksandrov) [reply WooFoundation/WooFoundation/Currency/CurrencySettings.swift:428]: It's actually not custom, but required because of `@Published` property preventing auto-synthesized conformance.
Agree about a comment, added in 9abe891

File: WooCommerce/StoreWidgets/StoreInfoProvider.swift (hunk 3 above)
Comment (ealeksandrov) [reply WooCommerce/StoreWidgets/StoreInfoProvider.swift]: Yes, updated in 37cca09

File: WooCommerce/StoreWidgets/StoreInfoProvider.swift (hunk 4 above)
Comment (ealeksandrov) [reply WooCommerce/StoreWidgets/StoreInfoProvider.swift:160]: Yeah, very obvious error, converted rate value to percentage twice 😄
Fixed in 72e118c.

**Comment by ealeksandrov** [comment]
@Ecarrion thanks for useful feedback and catching conversion bug!
We can't add unit test target to an extension, so I'll experiment about using our extension target in app unit tests bundle.

**Review by Ecarrion — APPROVED** [review]
Looks great!

I merged the timezone branch into this one to be able to test both changes at once. Also it helped resolving conflicts!
